articles/quickstart/webapp/python/01-login.md

```diff
@@ -1,118 +1,155 @@
+This handler exchanges the `code` that Auth0 sends to the callback URL and exchange for an `access_token` 
+and an `id_token`.
 
-${snippet(meta.snippets.setup)}
+The `id_token` is a [JWT](/jwt) that contains the information of the user profile, to get the information from it you have to decode and validate its signature. After the user information is obtained, store then in the flask `session`.
 
-## Access User Information
+```python
+# Here we're using the /callback route.
+@app.route('/callback')
+def callback_handling():
+    # Handles response from token endpoint
+    resp = auth0.authorized_response()
+    if resp is None:
+        raise Exception('Access denied: reason=%s error=%s' % (
+            request.args['error_reason'],
+            request.args['error_description']
+        ))
+    
+    # Obtain JWT and the keys to validate the signature
+    idToken = resp['id_token']
+    jwks = urlopen("https://"+${account.namespace}+"/.well-known/jwks.json")
+    
+    payload = jwt.decode(idToken, jwks.read(), algorithms=['RS256'], audience=${account.clientId},
+                        issuer="https://"+${account.namespace}+"/")
+    
+    # Store the tue user information obtained in the id_token in flask session.
+    session[constants.JWT_PAYLOAD] = payload
+    
+    session[constants.PROFILE_KEY] = {
+        'user_id': payload['sub'],
+        'name': payload['name'],
+        'picture': payload['picture']
+    }
+    
+    return redirect('/dashboard')
+```
 
-You can access the user information via the `profile` you stored in the session on step 2
+## Trigger Authentication
 
-```python
-@app.route("/dashboard")
-@requires_auth
-def dashboard():
-    return render_template('dashboard.html', user=session['profile'])
+Add a `/login` route that uses `Flask-OAuthlib` client instance to redirect the user to Auth0's hosted login page.
 
+```python
+@app.route('/login')
+def login():
+    return auth0.authorize(callback='${account.callback}')
 ```
 
+Create a `home.html` file in a `/template` folder. Add a link to the `/login` route.
+
 ```html
-<div>
-  <img class="avatar" src="{{user['picture']}}"/>
-  <h2>Welcome {{user['nickname']}}</h2>
+<div class="login-box auth0-box before">
+    <img src="https://i.cloudup.com/StzWWrY34s.png" />
+    <h3>Auth0 Example</h3>
+    <p>Zero friction identity infrastructure, built for developers</p>
+    <a class="btn btn-primary btn-lg btn-login btn-block" href="/login">Log In</a>
 </div>
 ```
 
-[Click here](/user-profile) to check all the information that the userinfo hash has.
-
 ## Logout
 
-You can implement logout by clearing a session and redirecting to [logout endpoint](/logout#redirect-users-after-logout).
+To logout the user you have to clear the data from the session, and redirect the user to the Auth0 logout endpoint.
+
+Checkout [logout documentation](/logout) for more information.
```

Comment:
To log the user out, you have to clear the data from the session and redirect to Auth0's logout endpoint. You can find more information about this in [our logout documentation](link to logout doc)